### PR TITLE
Add `devcontainer` configuration

### PR DESCRIPTION
**Pull request by ran-dall**
This adds the configuration for VSCode `Devcontainer` and Codespaces.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# See here for image contents: https://github.com/microsoft/vscode-dev-containers/tree/v0.241.1/containers/go/.devcontainer/base.Dockerfile
+
+# [Choice] Go version (use -bullseye variants on local arm64/Apple Silicon): 1, 1.18, 1.17, 1-bullseye, 1.18-bullseye, 1.17-bullseye, 1-buster, 1.18-buster, 1.17-buster
+ARG VARIANT="1.18-bullseye"
```
Comment (natemoo-re) [inline .devcontainer/Dockerfile:4]: I think `1.17` would be the safer option until we update our CI scripts and CONTRIBUTING instructions

Comment (ran-dall) [reply .devcontainer/Dockerfile:4]: @natemoo-re Thanks for the review. Appreciate it. I'm pushing it now. 

FWIW, in the future, I'd like to engineer something a little cleaner. Just sticking to the VSCode recommendations for now though.

**Comment by ran-dall** [comment]
@natemoo-re Before this is ready, I think I want to get rid of `nvm` in favor of `volta` that you're already using.

**Comment by ran-dall** [comment]
@natemoo-re Restarting the work on this PR. So I replaced Volta, but you still need to run `volta install node && npm install --global pnpm` to get everything set up. Maybe after https://github.com/volta-cli/volta/pull/1273, things will get nicer, but till then, I think this is as good as it's going to get.

**Comment by ran-dall** [comment]
@natemoo-re Actually, I fixed it. Now it will automatically set up `node` and `pnpm@7.0.1` with `volta` automatically. 🎉

**Pull request merged**